hadoop-hdds/common/src/test/java/org/apache/hadoop/hdds/protocol/MockDatanodeDetails.java

```diff
@@ -118,11 +128,21 @@ public static DatanodeDetails randomLocalDatanodeDetails()
       return createDatanodeDetails(UUID.randomUUID().toString(),
           socket.getInetAddress().getHostName(),
           socket.getInetAddress().getHostAddress(), null,
-          socket.getLocalPort());
+          getFreePort());
```

Comment:
Since we get the port from other source, we can get rid of the `ServerSocket` allocation.  Replace hostname and address with `"localhost"`, and `"127.0.0.1"`.

With that, `throws IOException` can be removed, as well as unused imports.

Reply:
Thanks for the suggestion!
Done!

Reply:
Done!